Cowles.js

```diff
@@ -0,0 +1,316 @@
+{
+	"translatorID": "951c027d-74ac-47d4-a107-9c3069ab7b48",
+	"label": "Cowles",
+	"creator": "Vincent Carret",
+	"target": "^https?://(www\\.)?cowles.yale.edu/",
+	"minVersion": "3.0.4",
+	"maxVersion": "",
+	"priority": 320,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2019-11-21 12:23:10"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2019 Vincent Carret
+	
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc, url) {
+	// TODO: adjust the logic here
+	if (url.includes('/cfp/')) {
+		return "journalArticle";
+	}
+	else if (url.includes('/cfdp/')) {
+		return "report";
+	}
+	else if (url.includes('/cfm-')) {
+		return "book";
+	}
+	else if (getSearchResults(doc, true)) {
+		return "multiple";
+	}
+	return false;
+}
+
+function getSearchResults(doc, checkOnly) {
+	var root = "https://cowles.yale.edu";
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll(".views-table tbody tr");
+	var href = "";
+	var title = "";
+	for (let row of rows) {
+		var paper = row.querySelector("td")
+		.textContent
+		.trim()
+		.toLowerCase()
+		.replace(".", "")
+		.split(" ");
+		// There are three types of documents: monographs (books), discussion papers (reports) and papers which were published
+		// This query method works on the monograph pages and the authors' pages for now
+		if (paper[0] == "cfm") {
+			href = root + "/" + paper.join("-");
+			title = row.querySelector("td.views-field-field-paper-title a").textContent;
```

Comment:
the `text` function simplifies this a good bit.

Reply:
I'm not sure I understand? (I'm really not proficient in js as you would have gathered)

Reply:
Implement `text` as a function in the beginning of the translator (see https://www.mediawiki.org/wiki/Citoid/Creating_Zotero_translators#text for an explanation and https://github.com/zotero/translators/commit/2901ba9fa5b4158cf87697a979a55ee7ad867fef#diff-1dbee0bc07ad76ab1291e9571884f32bR39 for an example), then do 
`title = text(row, "td.views-field-field-paper-title a");`